handlers.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/alphagov/paas-metrics/pkg/logcache"
+	"github.com/alphagov/paas-metrics/pkg/metric"
+	"github.com/sirupsen/logrus"
+)
+
+func (s *server) handleMetrics() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var wg sync.WaitGroup
+		m := []*metric.Metric{}
+		client := logcache.NewClient(s.logCacheAPI, r.Header.Get("Authorization"))
+
+		meta, err := client.Meta()
+		if err != nil {
+			s.logger.Error(err)
+			s.error(w, http.StatusInternalServerError, "we're experiencing issues, sorry")
+			return
+		}
+
+		// TODO: Consider optimising that. It runs fairly slow (almost 4s) with
+		// 745 instances running on my admin account :troll:
+		for sourceID := range meta.Meta {
```

Comment:
I believe the shard_groups API is meant for this purpose (you create a shard-group for the ids you want, then request the results from that group) ... but I think maybe that can be an improvement for later

Reply:
Yeah, I did consider shards but came to conclusion I didn't have reliable way of naming them. Maybe should pull something from JWT token? That involves decoding though.

I thought to leave it up for later but I don't have strong feelings. Your call.

Reply:
Also, there would be overhead trying to keep the shards up to date ;/